File: conformance/fixtures/javascript/src/calc.js
// The JavaScript half of the same deliberately small target: every line is
// mutable, and every line is tested. Kept line-for-line parallel to the Python
// fixture so a difference in the report is a difference in the FRAMEWORK.

function clamp(value, low, high) {
  if (value < low) return low;
  if (value > high) return high;
  return value;
}

function score(hits, total) {
  if (total === 0) return 0.0;
  return hits / total;
}

function tally(values) {
  let out = 0;
  for (const v of values) {
    out = out + v;
  }
  return out;
}

module.exports = { clamp, score, tally };

File: conformance/fixtures/java/src/main/java/calc/Calc.java
package calc;

/** The Java half of the same deliberately small target, kept parallel to the
 *  Python and JavaScript fixtures so a difference in the report is a difference
 *  in the FRAMEWORK rather than in the code under test. */
public final class Calc {

    private Calc() {}

    public static int clamp(int value, int low, int high) {
        if (value < low) {
            return low;
        }
        if (value > high) {
            return high;
        }
        return value;
    }

    public static double score(int hits, int total) {
        if (total == 0) {
            return 0.0;
        }
        return (double) hits / total;
    }

    public static int tally(int[] values) {
        int out = 0;
        for (int v : values) {
            out = out + v;
        }
        return out;
    }
}
